pkg/storegateway/series_chunks.go

```diff
@@ -58,8 +60,13 @@ func newSeriesChunksSeriesSet(from seriesChunksSetIterator) storepb.SeriesSet {
 	}
 }
 
-func newSeriesSetWithChunks(ctx context.Context, chunkReaders chunkReaders, chunksPool pool.Bytes, batches seriesChunkRefsSetIterator, stats *safeQueryStats) storepb.SeriesSet {
-	return newSeriesChunksSeriesSet(newPreloadingSetIterator[seriesChunksSet](ctx, 1, newLoadingSeriesChunksSetIterator(chunkReaders, chunksPool, batches, stats)))
+func newSeriesSetWithChunks(ctx context.Context, chunkReaders chunkReaders, chunksPool pool.Bytes, batches seriesChunkRefsSetIterator, stats *safeQueryStats, iteratorLoadDurations *prometheus.HistogramVec) storepb.SeriesSet {
+	var iterator seriesChunksSetIterator
+	iterator = newLoadingSeriesChunksSetIterator(chunkReaders, chunksPool, batches, stats)
+	iterator = newDurationMeasuringIterator[seriesChunksSet](iterator, iteratorLoadDurations.WithLabelValues("chunks_load"))
+	iterator = newPreloadingSetIterator[seriesChunksSet](ctx, 1, iterator)
+	iterator = newDurationMeasuringIterator[seriesChunksSet](iterator, iteratorLoadDurations.WithLabelValues("chunks_preloaded"))
```

Comment:
The tracking here will test us how long we wait for the preloader. Ideally this should be 0. It won't be zero, but let's say the lower the better, while we don't care that much about "chunks_load" as far as "chunks_preloaded" is very low. Am I correct?

If so, I think we should mention this somewhere. Maybe here as a comment.

Reply:
> we don't care that much about "chunks_load" as far as "chunks_preloaded" is very low

you are correct. I was thinking we can use this to see if we need to introduce any parallelism in loading the chunks (e.g. two batches at a time)

I will add this info to the help text

Reply:
oh yeah. i can't add it to the help text of the whole metric 🤦 I will add a comment